### ISSUE TITLE
hello thank you for your work on this problem, i hope this is the right spot to ask for help

### ISSUE THREAD

**Issue by forestsounds89** [CheariX/silverblue-akmods-keys#4]
thank you for your work on this problem, when i run the setup.sh i get this error: cp: cannot stat 'macros.kmodtool': No such file or directory
not sure whats wrong and will greatly appreciate your help

i posted about this here as well but was told not too so here i am, https://github.com/fedora-silverblue/issue-tracker/issues/272

**Comment by CheariX** [comment]
Hi,

yes - I guess this is the right place :-)

Unfortunately, I cannot really reproduce your issue.
The following works for me (on Silverblue 36 and 37):

```sh
git clone https://github.com/CheariX/silverblue-akmods-keys 
cd silverblue-akmods-keys/ 
sudo sh setup.sh  
rpm-ostree install akmods-keys-0.0.2-8.fc37.noarch.rpm 
```

Does this work for you?
Did you checkout the whole projekt via git?
Do you run `setup.sh` with `sudo` ?

**Comment by forestsounds89** [comment]
thank you very much for reply, I might need to reinstall silverblue and try this again in the correct order
no that did not work for me 
i have read the project but i am still learning bash and git and silverblue, i am a fast learner 
yes i ran setup.sh as sudo

from a fresh install of silver blue with full disk encryption and secure boot enabled what steps should i follow, this is what ive done so far:
enable 3rd party repos during install
do updates after first boot and reboot
sudo rpm-ostree install https://mirrors.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm https://mirrors.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
reboot then:
sudo rpm-ostree install akmod-nvidia xorg-x11-drv-nvidia && sudo rpm-ostree kargs --append=rd.driver.blacklist=nouveau --append=modprobe.blacklist=nouveau --append=nvidia-drm.modeset=1

from here steps are not clear to me how to run these, i also seem to have forgotten the steps i followed to sign the keys the first time i was following workstation advice not silverblue and i have gotten confused in my learning and memory and feel a bit lost all of a sudden, thank you very much for the help

rpm-ostree install rpmdevtools
kmodgenca 
mokutil --import /etc/pki/akmods/certs/public_key.der    

bash setup.sh
rpm-ostree install akmods-keys-0.0.2-8.fc$(rpm -E %fedora).noarch.rpm

**Comment by forestsounds89** [comment]
from a fresh install to eliminate some variables, what steps do take before running these commands?:
git clone https://github.com/CheariX/silverblue-akmods-keys 
cd silverblue-akmods-keys/ 
sudo sh setup.sh  
rpm-ostree install akmods-keys-0.0.2-8.fc37.noarch.rpm

**Comment by CheariX** [comment]
Actually, everything you described looks fine to me.

- You need to enable RPM Fusion. Install `akmod-nvidia` and configure `kargs`.  AFAIK, you do not need `xorg-x11-drv-nvidia ` as it is a dependency for `akmod-nvidia`. Your steps described above look good for me.
- `rpm-ostree install rpmdevtools` -> Reboot
- Configure `kmodgenca`. I found [this link](https://nelsonaloysio.medium.com/how-to-sign-nvidia-proprietary-driver-to-secure-boot-fedora-centos-rhel-6da37db0ff41) very helpful.
- and then do the steps from your previous comment.

In summary, I find no fault in your approach :(

Also, your actual error message `cp: cannot stat 'macros.kmodtool'` does not look like a `rpm-ostree`-related issue.

**Comment by forestsounds89** [comment]
thank you so much for you help, i wanted to post back and let you know everything worked as expected with 37 beta on my system, 36 i ran into trouble and im not sure if it was my mistake or not so im gonna do another test with 36 see if i can learn anything, ive learned alot in the last few days my brain is sore lol, any suggestions on where to start my learning path?, so far i have found free courses for differnt types of code, ssh, bash, but have not started yet, i feel like this is the future so i want to learn more about how to use fedora immutable OS, its been years since i used windows but im still not a great linux user, its time to get serious :)
